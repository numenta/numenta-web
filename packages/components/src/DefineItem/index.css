/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 65%;
}

.vertical {
  composes: mb3 from 'tachyons';
  composes: ml0 from 'tachyons';
  composes: pl0 from 'tachyons';
}


.defineItem {
  word-wrap: break-word;
}
