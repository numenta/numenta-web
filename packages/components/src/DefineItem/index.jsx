// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const DefineItem = ({children, direction}) => {
  const classes = [styles.defineItem]

  if (direction) classes.push(styles[direction])

  return (
    <dd className={classNames(...classes)}>
      {children}
    </dd>
  )
}

DefineItem.propTypes = {
  children: React.PropTypes.node.isRequired,
  direction: React.PropTypes.oneOf(['horizontal', 'vertical']),
}

export default DefineItem
