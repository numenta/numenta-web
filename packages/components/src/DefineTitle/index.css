/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 35%;
}

.vertical {
  composes: lh-title from 'tachyons';
  composes: mb1 from 'tachyons';
}


.defineTitle {
  composes: fw6 from 'tachyons';
}
