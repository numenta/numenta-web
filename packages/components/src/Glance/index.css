/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._rules dl > dt,
._rules dl > dd {
  border-top: 1px solid #ddd;
}

.glance {
  composes: bg-near-white from 'tachyons';
  composes: br2 from 'tachyons';
  composes: f5 from 'tachyons';
  composes: f6-ns from 'tachyons';
  composes: mb2 from 'tachyons';
  composes: pb3 from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: pt2 from 'tachyons';
  composes: tl from 'tachyons';
  composes: _rules;
}
