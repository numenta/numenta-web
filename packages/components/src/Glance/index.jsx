// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import DefineList from '../DefineList'

import styles from './index.css'


/**
 * At-A-Glance React view component. Quick Details about the company/site.
 */
const Glance = ({children, head}) => (
  <aside className={styles.glance}>
    <div>{head}</div>
    <DefineList direction="horizontal">
      {children}
    </DefineList>
  </aside>
)

Glance.propTypes = {
  children: React.PropTypes.array.isRequired,
  head: React.PropTypes.element.isRequired,
}

export default Glance
