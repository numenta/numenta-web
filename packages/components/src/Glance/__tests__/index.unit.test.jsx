import React from 'react'
import renderer from 'react-test-renderer'
import stubContext from 'react-stub-context'

import context from '__tests__/__mocks__/reactContextMock'

import DefineItem from '../../DefineItem'
import DefineTitle from '../../DefineTitle'
import Glance from '../../Glance'


describe('Glance React component', () => {

  it('Renders correctly', () => {
    const GlanceStubbed = stubContext(Glance, context)
    const component = renderer.create(
      <GlanceStubbed head={<h2>Glance Header</h2>}>
        <DefineTitle>
          Title
        </DefineTitle>
        <DefineItem>
          Definition
        </DefineItem>

        <DefineTitle>
          Another
        </DefineTitle>
        <DefineItem>
          Last
        </DefineItem>
      </GlanceStubbed>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
