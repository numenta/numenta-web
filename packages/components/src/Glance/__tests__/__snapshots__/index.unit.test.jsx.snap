exports[`Glance React component Renders correctly 1`] = `
<aside
  className="glance">
  <div>
    <h2>
      Glance Header
    </h2>
  </div>
  <dl
    className="defineList horizontal">
    <dt
      className="defineTitle horizontal">
      Title
    </dt>
    <dd
      className="defineItem horizontal">
      Definition
    </dd>
    <dt
      className="defineTitle horizontal">
      Another
    </dt>
    <dd
      className="defineItem horizontal">
      Last
    </dd>
  </dl>
</aside>
`;
